### PR TITLE
Fix issue 858

### PR DESCRIPTION
**Pull request by ejalaa12**
**Public API Changes**

None

**Description**

> When using a Service Client, and sending two request in a row, because the Client saves one callback at a time, only the latest is used when processing the response message from the service.

fix issue https://github.com/RobotWebTools/rclnodejs/issues/858

### CODE REVIEW

File: lib/client.js [hunk 1]
```diff
@@ -28,7 +28,7 @@ class Client extends Entity {
     super(handle, typeClass, options);
     this._nodeHandle = nodeHandle;
     this._serviceName = serviceName;
-    this._sequenceNumber = 0;
+    this._pending_requests = new Map();
```
Comment (minggangw) [inline lib/client.js]: I would suggest renaming it to `this._senquenceNumberToCallbackMap` to align with the JS naming convention

File: lib/node.js [hunk 2]
```diff
@@ -204,13 +204,12 @@ class Node extends rclnodejs.ShadowNode {
       this._runWithMessageType(
         client.typeClass.Response,
         (message, deserialize) => {
-          let success = rclnodejs.rclTakeResponse(
+          let sequence_number = rclnodejs.rclTakeResponse(
```
Comment (minggangw) [inline lib/node.js]: Please rename it to `sequenceNumber`

File: src/rcl_bindings.cpp [hunk 3]
```diff
@@ -831,24 +831,21 @@ NAN_METHOD(RclTakeResponse) {
...
-      reinterpret_cast<rmw_request_id_t*>(malloc(sizeof(rmw_request_id_t)));
-  header->sequence_number = sequence_number;
+  rmw_service_info_t header;
 
   void* taken_response =
-      node::Buffer::Data(Nan::To<v8::Object>(info[2]).ToLocalChecked());
-  rcl_ret_t ret = rcl_take_response(client, header, taken_response);
-  free(header);
+      node::Buffer::Data(Nan::To<v8::Object>(info[1]).ToLocalChecked());
+  rcl_ret_t ret = rcl_take_response_with_info(client, &header, taken_response);
```
Comment (minggangw) [inline src/rcl_bindings.cpp:839]: `rcl_take_response_with_info()` was not there the time when we implemented this JS function binding, thanks for updating it! I believe some other rcl functions are deprecated by the cpp/py clients already, which we need to pay attention to.

**Review by minggangw — COMMENTED** [review]
Thanks for fixing this quickly, some comments are below.

File: lib/client.js (hunk 1 above)
Comment (ejalaa12) [reply lib/client.js]: Thanks for pointing this out ! I just pushed a commit with the change

**Review by minggangw — APPROVED** [review]
Thanks for fixing this issue! lgtm and merging. Your contributions are always welcome :)

**Pull request merged**